Source/artifacts/ArtifactsManager.js

```diff
@@ -39,68 +39,60 @@ export class ArtifactsManager {
+        _inquirerManager.get(this).promptForCommand(language)
+            .then(context => {
+                this._logger.info(`Creating command with name '${context.name}' and namespace '${context.namespace}'`);
+                _boilerPlatesManager.get(this).createArtifactInstance('command', language, boilerPlate, destination, context);  
+            });
     }
     /**
      * Create an event
      * @param {string} name 
      * @param {string} namespace 
      */
-    createEvent(name, namespace) {
-        this._logger.info(`Creating event with name '${name}' and namespace '${namespace}'`);
-
+    createEvent(language) {
+        
         let boilerPlate = _boilerPlatesManager.get(this).boilerPlatesByType('artifacts')[0];
-        let context = {
-            name: name,
-            namespace: namespace
-        };
         let destination = process.cwd();
-        
-        _boilerPlatesManager.get(this).createArtifactInstance('event', 'csharp', boilerPlate, destination, context);
+        _inquirerManager.get(this).promptForEvent(language)
+            .then(context => {
+                this._logger.info(`Creating event with name '${context.name}' and namespace '${context.namespace}'`);
+                _boilerPlatesManager.get(this).createArtifactInstance('event', language, boilerPlate, destination, context);  
+            });
     }
     /**
      * Create a read model
      * @param {string} name 
      * @param {string} namespace 
      */
-    createReadModel(name, namespace) {
-        this._logger.info(`Creating read model with name '${name}' and namespace '${namespace}'`);
+    createReadModel(language) {
 
         let boilerPlate = _boilerPlatesManager.get(this).boilerPlatesByType('artifacts')[0];
-        let context = {
-            name: name,
-            namespace: namespace
-        };
         let destination = process.cwd();
-        
-        _boilerPlatesManager.get(this).createArtifactInstance('readModel', 'csharp', boilerPlate, destination, context);
+        _inquirerManager.get(this).promptForReadModel(language)
+            .then(context => {
+                this._logger.info(`Creating read model with name '${context.name}' and namespace '${context.namespace}'`);
+                _boilerPlatesManager.get(this).createArtifactInstance('readModel', language, boilerPlate, destination, context);  
+            });
     }
     /**
      * Create an aggregate root
      * @param {string} name 
      * @param {string} namespace 
      */
-    createAggregateRoot(name, namespace) {
-        this._logger.info(`Creating aggregate root with name '${name}' and namespace '${namespace}'`);
+    createAggregateRoot(language) {
 
         let boilerPlate = _boilerPlatesManager.get(this).boilerPlatesByType('artifacts')[0];
-        let context = {
-            name: name,
-            namespace: namespace
-        };
         let destination = process.cwd();
```

Comment:
Using process directly will make automated specs virtually impossible to write consistently. I suggest introducing a context object that can can be taken dependencies on and set up during globals setup. In it we could have the current working directory as a propery.